receiver/zookeeperreceiver/factory.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package zookeeperreceiver
+
+import (
+	"context"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configmodels"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/receiver/receiverhelper"
+)
+
+const (
+	typeStr = "zookeeper"
+
+	defaultCollectionInterval = 10 * time.Second
+	defaultConnectionTimeout  = 10 * time.Second
+)
+
+func NewFactory() component.ReceiverFactory {
+	return receiverhelper.NewFactory(
```

Comment:
Consider to also look at https://github.com/open-telemetry/opentelemetry-collector/blob/master/receiver/receiverhelper/scraper.go to use that

Reply:
Updated the config to use helper from the `receiverhelper` package. Will also make use of the helpers for the data collection logic.

Reply:
For a single scraper like this is, I don't see what value the scraper helper offers other than the timed interval loop, but this can be reduced down to a function that takes a context and a time interval and function (the equivalent of `Scrape`), just like the [RunOnInterval](https://github.com/signalfx/signalfx-agent/blob/master/pkg/utils/time.go#L45) function in the Smart Agent.  The receiver itself already has the notions of `Initialize` and `Close` but it calls it `Start` and `Shutdown` so it is just adding another layer of complexity with different terminology that has no value.  Once I get my simple metric helper in then the metric facilities in the scraper thing will be obsolete as well.  Also it decouples contrib code from helpers in core that are hard to change.

Reply:
@keitwb - yes, I agree functionality-wise there is no added value in using the scraper helper in this particular case. On the other hand, I think it provides a consistent implementation across receivers. That said, I also agree about this logic being reduced to a single function. I can refactor to make it such that it could be used with a `RunOnInterval` equivalent and just use the receiver struct rather than adding the scraper layer. Does that sound good to you?

Reply:
Please continue to use the Scraper framework. One of the advantage is that it can batch scrapers so limits the amount of timers (@keitwb not only the loop), and we are planning to develop more the scraper framework, even consider making it part of the service.